### ISSUE TITLE
vncpasswd and vncviewer common passwd file handling: regression or undocumented behavior change 

### ISSUE THREAD

**Issue by MichaelRiss** [TigerVNC/tigervnc#1821]
**Describe the bug**
I see a change in behavior between TigerVNC 1.13.X and 1.14.X on Fedora.
On 1.13.X you could provide a main password and a guest password to `vncpasswd` and hand the resulting `passwd` file over to `vncviewer` and `vncviewer` was able to correctly extract the main password from the file.
With 1.14.X `vncviewer` chokes on such a combined `passwd` file and shows an error window:
![vncviewer-bad-password-length](https://github.com/user-attachments/assets/1e3288ee-cf2f-4f40-bb08-482df0fcd1ca)
And on the console you get this output - note the `bad obfuscated password length` part:
```
TigerVNC Viewer v1.14.0
Built on: 2024-08-05 00:00
Copyright (C) 1999-2024 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.

Thu Aug 22 22:25:15 2024
 DecodeManager: Detected 24 CPU core(s)
 DecodeManager: Creating 4 decoder thread(s)
 CConn:       Connected to host localhost port 56249
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 CConnection: Choosing security type VeNCrypt(19)
 CVeNCrypt:   Choosing security type TLSVnc (258)
 CConn:       bad obfuscated password length
 DecodeManager:   Total: 0 rects, 0 pixels
 DecodeManager:          0 B (1:-nan ratio)
```

If you type in the main password manually - it works.
If you do not provide a guest password to `vncpasswd` - `vncviewer` can use the resulting `passwd` file to log into the session.

**To Reproduce**
1. Use `vncpasswd` and provide main password *and* guest password
2. Start a VNC session with this password file
3. Use the `passwd` file with the `-PasswordFile` option in `vncviewer` to attempt to log into the session
4. It will fail
5. Try the same *without* providing the guest password - it should work
6. Do not use the `-PasswordFile` option in `vncviewer` and enter the password manually - should also work

**Expected behavior**
Either re-establish the old behavior so that `vncviewer` can handle combined `passwd` files or update the documentation of `vncviewer` to point out that the `passwd` file must not contain a guest password or reading it will fail.

**Screenshots**
See above in the main description.

**Client (please complete the following information):**
 - OS: Fedora 40
 - VNC client: TigerVNC
 - VNC client version: 1.13.X now updated to 1.14.X
 - Client downloaded from: Fedora 40 main repos

**Server (please complete the following information):**
 - OS: Fedora 40
 - VNC server: TigerVNC
 - VNC server version: 1.13.X now updated to 1.14.X
 - Server downloaded from: Fedora 40 main repos
 - Server was started using: `/usr/bin/systemctl start vncserver@:${screennumber}.service` (${screennumber} is a variable depending on the user who wants to start a session on the system)

**Comment by CendioOssman** [comment]
Ah. This was not a scenario that was considered, since vncviewer can only use a single password.

Could you give some more context on why it is convenient for you to have a passwd file with two passwords for the client, even though just one can be used?

**Comment by MichaelRiss** [comment]
We have a script for our company. It did basically the following:
- Log into the remote server and generate two random passwords, one main password, one guest password, start vncserver with both.
- Copy back the passwd file to the client machine in order to start vncviewer.
- But also print the guest password on the console in case the user wants to invite a colleague to observe a work procedure on the remote server.

In the end the guest option never got adopted as conference tool screen sharing is quite convenient nowadays. So for me it was just a sudden behavior change which took a while to figure out. So I have no problem with dropping the feature, but maybe document it?

**Comment by CendioOssman** [comment]
It sounds reasonable that more might have and automated deployment like you guys, so let me see if I can get some symmetry in what the server and client tolerate.

**Comment by CendioOssman** [comment]
Should be fixed in 49a087e8f8ecafe950a86a71ae6657d6cac8ea77.
